hw/ip/otbn/util/shared/isr.py

```diff
@@ -0,0 +1,54 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Iterable
+from serialize.parse_helpers import (check_keys, check_str, check_list,
+                                     check_int, check_bool, load_yaml)
+
+
+class Isr:
+    def __init__(self, name: str, address: int,
+                 doc: str, read_only: bool, bits: Dict[int, str]) -> None:
+        self.name = name
+        self.address = address
+        self.doc = doc
+        self.read_only = read_only
+        self.bits = bits
+
+    def access_str(self) -> str:
+        return 'RO' if self.read_only else 'RW'
+
+    @staticmethod
+    def from_yml(yml: object) -> 'Isr':
+        yd = check_keys(yml, 'wsr',
```

Comment:
This is also used for CSRs would it be worth using something more general for the `what`, instead of 'wsr'?